### DISCUSSION TITLE
A database error

### DISCUSSION THREAD

**Discussion by YiFANzZhou** [WrightonLabCSU/DRAM#178] [Q&A] [answered]
Hi, I am trying to use DRAM-v to annotate my viral contigs.

after $ DRAM-setup.py prepare_databases --output_dir yifan/DRAM/DRAM_DBs

there is a warning:

Database preparation started
Downloading dbCAN family activities from : https://bcb.unl.edu/dbCAN2/download/Databases/V10/CAZyDB.07292021.fam-activities.txt
Downloading dbCAN from: http://bcb.unl.edu/dbCAN2/download/dbCAN-HMMdb-V10.txt
0:00:07.547237: dbCAN database processed
1:12:20.159645: UniRef database processed
1:46:45.797195: PFAM database processed
1:46:56.969548: RefSeq viral database processed
1:47:18.301355: MEROPS database processed
1:49:40.240174: VOGdb database processed
1:56:32.679999: KOfam database processed
1:56:37.641060: KOfam ko list processed
1:56:37.641100: PFAM hmm dat processed
1:56:37.641108: dbCAN fam activities processed
1:56:37.795934: VOGdb annotations processed
1:56:38.924671: DRAM databases and forms downloaded
1:56:39.060705: Files moved to final destination
/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/database_handler.py:51: UserWarning: Database does not exist at path None
  warnings.warn('Database does not exist at path %s' % self.description_loc)
1:56:39.181648: DRAM description database populated
1:58:18.846037: Database preparation completed

when I try to run a DRAM-v job, it returns an error :

/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/database_handler.py:81: UserWarning: No descriptions were found for your id's. Does this YP_009174077.1 look like an id from viral_description

  warnings.warn("No descriptions were found for your id's. Does this %s look like an id from %s" % (list(ids)[0],
2022-05-17 11:31:53.289489: Viral annotation started
0:00:00.012122: Retrieved database locations and descriptions
0:00:00.012143: Annotating H1_viral_contigs
0:01:01.633608: Turning genes from prodigal to mmseqs2 db
0:01:03.189002: Getting hits from kofam
0:23:00.128023: Getting forward best hits from viral
0:23:24.844598: Getting reverse best hits from viral
0:23:27.923433: Getting descriptions of hits from viral
Traceback (most recent call last):
  File "/homes/yizhou/miniconda3/envs/DRAM/bin/DRAM-v.py", line 153, in <module>
    args.func(**args_dict)
  File "/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/annotate_vgfs.py", line 475, in annotate_vgfs
    annotations = annotate_fastas(contig_locs, output_dir, db_handler, min_contig_size, prodigal_mode, trans_table,
  File "/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/annotate_bins.py", line 1013, in annotate_fastas
    annotate_fasta(fasta_loc, fasta_name, fasta_dir, db_handler, min_contig_size, prodigal_mode, trans_table,
  File "/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/annotate_bins.py", line 921, in annotate_fasta
    annotations = annotate_orfs(gene_faa, db_handler, tmp_dir, start_time, custom_db_locs, custom_hmm_locs,
  File "/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/annotate_bins.py", line 814, in annotate_orfs
    annotation_list.append(do_blast_style_search(query_db, db_handler.db_locs['viral'], tmp_dir,
  File "/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/annotate_bins.py", line 684, in do_blast_style_search
    hits = formater(hits, header_dict)
  File "/homes/yizhou/miniconda3/envs/DRAM/lib/python3.10/site-packages/mag_annotator/annotate_bins.py", line 173, in get_basic_description
    header = header_dict[hit]
KeyError: 'YP_009174077.1'

Could you please help me to fix this?

Looking forward to your favorable reply

Best,

Yifan

**Comment by rmFlynn** [comment, marked as answer]
Can you try running `DRAM-setup.py update_description_db` and see if that solves the problem, usually this error is from something going wrong during setup, but in this case viral may have made a change to the database that needs fixing. I will look really quick

**Reply by YiFANzZhou** [reply]
thank you! ! !

after  **$ DRAM-setup.py update_description_db** 

DRAM-v works pretty well.
